src/components/JaegerToolbar/LookBack.tsx

```diff
@@ -0,0 +1,130 @@
+  }
+
+  render() {
+    const { lookback, fetching, setLookback, onChangeCustom } = this.props;
+    const tz = lookback === 'custom' ? new Date().toTimeString().replace(/^.*?GMT/, 'UTC') : null;
+
+    return (
+      <span style={{ marginLeft: '10px' }}>
+        <Col componentClass={Form.ControlLabel} style={{ marginRight: '10px' }}>
+          Lookback
+        </Col>
+        <ToolbarDropdown
+          id="lookback-selector"
+          disabled={fetching}
+          options={this.lookBackOptions}
+          value={lookback}
+          label={this.lookBackOptions[lookback]}
+          useName={false}
+          handleSelect={setLookback}
+        />
+        {tz && (
+          <Form style={{ display: 'inline-flex' }} inline={true}>
+            <FormGroup>
+              <Col componentClass={Form.ControlLabel}>
+                Start Time
+                <FieldLevelHelp
+                  style={{ marginLeft: '10px' }}
+                  content={<div>Times are expressed in {tz}</div>}
+                  placement={'bottom'}
+                />
+              </Col>
+
+              <FormControl
+                style={{ marginLeft: '10px' }}
+                type="date"
+                disabled={false}
+                onChange={e => onChangeCustom('start', e.target.value, '')}
+              />
+              <FormControl
+                style={{ marginLeft: '10px' }}
+                type="time"
+                disabled={false}
+                onChange={e => onChangeCustom('start', '', e.target.value)}
+              />
+            </FormGroup>
+            <FormGroup>
+              <Col componentClass={Form.ControlLabel}>
+                End Time
+                <FieldLevelHelp
+                  style={{ marginLeft: '10px' }}
+                  content={<div>Times are expressed in {tz}</div>}
+                  placement={'bottom'}
+                />
+              </Col>
+
+              <FormControl
+                style={{ marginLeft: '10px' }}
+                type="date"
+                disabled={false}
+                onChange={e => onChangeCustom('end', e.target.value, '')}
+              />
+              <FormControl
+                style={{ marginLeft: '10px' }}
+                type="time"
+                disabled={false}
+                onChange={e => onChangeCustom('end', '', e.target.value)}
+              />
+            </FormGroup>
+          </Form>
+        )}
+      </span>
+    );
+  }
+}
+
+const mapStateToProps = (state: KialiAppState) => {
+  return {
+    fetching: state.jaegerState.search.serviceSelected === '',
+    lookback: state.jaegerState.search.lookback
+  };
+};
+
+const mapDispatchToProps = (dispatch: ThunkDispatch<KialiAppState, void, KialiAppAction>) => {
+  return {
+    setLookback: (lookback: string) => {
+      dispatch(JaegerActions.setLookback(lookback));
+    }
+  };
+};
+
+const LookBackContainer = connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(LookBack);
+
```

Comment:
For example here @jotak 

What do you think about this @mtho11 ? I saw other components like https://github.com/kiali/kiali-ui/blob/3ec7c4d7048227e54d6523358bbc8888f86fca1e/src/components/GraphFilter/GraphFilter.tsx where we set mapStateToProps & mapDispatchToProps in the same file instead create a new file in /containers.

I think that could be cleaner if we set this in the same file, the developer can check this props in teh same file without having to go to containers folder and we would have fewer files

Reply:
+1